core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -58,11 +68,211 @@ case class PSBT(
       vec.head.version
     }
   }
+
+  /**
+    * @param tx Transaction to add to PSBT
+    * @param i index of inputMap to add tx to
+    * @return PSBT with added tx
+    */
+  def addTransaction(tx: Transaction, i: Int): PSBT = {
+    require(
+      i < inputMaps.size,
+      s"i must be less than the number of input maps present in the psbt, $i >= ${inputMaps.size}")
+    if (inputMaps(i).isFinalized) {
+      this
+    } else {
+      var elements = inputMaps(i).elements
+      val txIn = transaction.inputs(i)
+
+      if (txIn.previousOutput.vout.toInt < tx.outputs.size) {
+        val out = tx.outputs(txIn.previousOutput.vout.toInt)
+
+        val outIsWitnessScript =
+          WitnessScriptPubKey.isWitnessScriptPubKey(out.scriptPubKey.asm)
+        val hasWitScript =
+          inputMaps(i)
+            .getRecords[InputPSBTRecord.WitnessScript](WitnessScriptKeyId)
+            .size == 1
+
+        if (outIsWitnessScript || hasWitScript) {
+          elements = elements.filterNot(_.key.startsWith(hex"01")) :+ WitnessUTXO(
+            out)
+        } else {
+          elements = elements.filterNot(_.key.startsWith(hex"00")) :+ NonWitnessOrUnknownUTXO(
+            tx)
+        }
+      }
+
+      val newInputMaps =
+        inputMaps.patch(i, Seq(InputPSBTMap(elements)), 1)
+      PSBT(globalMap, newInputMaps, outputMaps)
+    }
+  }
+
+  /**
+    * @param script ScriptPubKey to add to PSBT
+    * @param i index of inputMap to add script to
+    * @return PSBT with added script
+    */
+  def addScriptToInput(script: ScriptPubKey, i: Int): PSBT = {
```

Comment:
similar to above, `i -> index` and would be nice to have a more descriptive scaladoc